discretisedfield/field.py

```diff
@@ -2508,22 +2509,31 @@ def div(self):
         >>> f.div.mean()
         array([5.])
 
-        2. Attempt to compute the divergence of a scalar field.
-
-        >>> f = df.Field(mesh, nvdim=1, value=3.14)
-        >>> f.div
-        Traceback (most recent call last):
-        ...
-        ValueError: ...
-
         .. seealso:: :py:func:`~discretisedfield.Field.derivative`
 
         """
-        if self.nvdim not in [2, 3]:
-            msg = f"Cannot compute divergence for nvdim={self.nvdim} field."
-            raise ValueError(msg)
+        if self.nvdim != self.mesh.region.ndim:
+            raise ValueError(
+                f"Cannot compute divergence for field with a differnt {self.nvdim=} and"
+                f" {self.mesh.region.ndim}."
+            )
 
-        return sum(getattr(self, vdim).diff(vdim) for vdim in self.vdims)
+        for vdim in self.vdims:
+            if vdim not in self.vdim_mapping:
+                raise ValueError(
+                    f"Cannot compute divergence for field as {self.vdim_mapping=} is"
+                    " not set properly."
+                )
+            else:
+                if self.vdim_mapping[vdim] not in self.mesh.region.dims:
+                    raise ValueError(
+                        "Cannot compute divergence for field as"
```

Comment:
Similarly, will it be good to mention that the associated geometrical dimension in the `vdim_mapping` is missing from the `region.dims` in the error message?

Reply:
I've added this as well!